### ISSUE TITLE
critic output nonlinearity, loss and `n_critic`

### ISSUE THREAD

**Issue by pavanramkumar** [martinarjovsky/WassersteinGAN#3]
thanks for your great work!

i'd like to implement a WGAN critic for `keras` but i'm new to torch and having a hard time figuring out a couple of things in your code.

- i want to double check what nonlinearity do you use for your critic output. i see that you use linear for your `MLP_D` [here](https://github.com/martinarjovsky/WassersteinGAN/blob/master/models/mlp.py#L50) but i don't see it in `DCGAN_D` or `DCGAN_D_nobn`.

- why linear if `f_w` is a distance metric? why not `relu`?

- am i right that this is the loss for the critic? (`y_true` is 0/1 for generated/training sample)
```python
def wgan_loss(y_true, y_pred):
    return K.mean(((-1) ** y_true) * y_pred)
```

- in practice, what value of `n_critic` do you use?

**Comment by LukasMosser** [comment]
@pavanramkumar maybe this can help you reproduce this in Keras:
[Keras-WGAN](https://github.com/tdeboissiere/DeepLearningImplementations/tree/master/WassersteinGAN/src/model)

**Comment by pavanramkumar** [comment]
@LukasMosser thanks! this helps!
